lib/utils/compose_ts.ts

```diff
@@ -411,6 +417,18 @@ async function installQemuIfNeeded({
 	return needsQemu;
 }
 
+export function makeImageName(
+	projectName: string,
+	serviceName: string,
+	tag?: string,
+) {
+	let name = `${projectName}_${serviceName}`;
+	if (tag) {
+		name = [name, tag].map((s) => s.replace(/:/g, '_')).join(':');
```

Comment:
I've added this bit of logic as I have learned that, in the past, [we advised users to add a colon to the project name](https://jel.ly.fish/support-thread-discourse-323654), e.g. `--projectName myproject:v1.1.1`, as a workaround for the fact that `--tag` didn't work, even though the service name was then appended to the tag, e.g. `myproject:v1.1.1_main`. This logic ensures a valid image name by replacing the colon with an underscore if `--tag` is used, thus preserving compatibility.